### PR TITLE
Test C# against 3.7 services in PATH

### PR DESCRIPTION
**Pull request by externl**

### CODE REVIEW

File: scripts/Glacier2Util.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import sys, os
 from Util import *
 
-class Glacier2Router(ProcessFromBinDir, ProcessIsReleaseOnly, Server):
+class Glacier2Router(ProcessIsFromPath, ProcessFromBinDir, ProcessIsReleaseOnly, Server):
```
Comment (pepone) [inline scripts/Glacier2Util.py:8]: Why do we keep ProcessFromBinDir here, is this temporary?

**Review by pepone — APPROVED** [review]
Looks good

File: scripts/Glacier2Util.py (hunk 1 above)
Comment (externl) [reply scripts/Glacier2Util.py:8]: We could probably remove it in a cleanup pass later.  I can remove it now if you rather?

Comment (pepone) [reply scripts/Glacier2Util.py:8]: If it is no longer require just remove it.

Comment (externl) [reply scripts/Glacier2Util.py]: ok, I removed it from the glacier2 and icegrid classes

File: csharp/test/Glacier2/router/Client.cs [hunk 2]
```diff
@@ -55,9 +55,8 @@ public override async Task RunAsync(string[] args)
             {
                 Console.Out.Write("getting the session timeout... ");
                 Console.Out.Flush();
-                long sessionTimeout = router.GetSessionTimeout();
-                long acmTimeout = router.GetACMTimeout();
-                Assert(sessionTimeout == 30 && acmTimeout == 30);
+                long timeout = router.GetSessionTimeout();
```
Comment (bentoi) [inline csharp/test/Glacier2/router/Client.cs:58]: Why not use the ACM timeout instead which is the preferred way now to keep alive the session? We'll eventually remove the old session timeout in future Ice 3.x versions.

**Review by bentoi — APPROVED** [review]
Looks ok but we'll need at some point to review our strategy for these scripts. Copying these files in each repository is going to become problematic for maintenance if we have many repositories.

Note also that DataStorm took another approach, see https://github.com/zeroc-ice/datastorm/blob/master/scripts/Component.py

However, it's a bit different I guess here since icerpc won't depend on Ice. It will just be available on the VM for testing purposes. That said... for interoperability testing we'll have to build client/server against the installed Ice 3.x version and we'll have 3.x test sources in the repository that depend on it?

**Pull request merged**